knowage-vue/src/modules/documentExecution/documentDetails/dialogs/olapDesignerDialog/DocumentDetailOlapDesignerDialog.vue

```diff
@@ -0,0 +1,168 @@
+                        <template #value="slotProps">
+                            <div v-if="slotProps.value">
+                                <span>{{ $t(`documentExecution.documentDetails.designerDialog.${slotProps.value}`) }}</span>
+                            </div>
+                        </template>
+                        <template #option="slotProps">
+                            <div>
+                                <span>{{ $t(slotProps.option.label) }}</span>
+                            </div>
+                        </template>
+                    </Dropdown>
+                    <label for="type" class="kn-material-input-label"> {{ $t('documentExecution.documentDetails.designerDialog.templateType') }}</label>
+                </span>
+            </div>
+        </div>
+        <DocumentDetailXMLAForm v-if="type === 'xmla'" class="p-m-4" :xmlModel="xmlModel"></DocumentDetailXMLAForm>
+        <DocumentDetailMondrianForm v-else class="p-m-4" :sbiExecutionId="sbiExecutionId" :mondrianModel="mondrianModel" :mondrianSchemas="mondrianSchemas"></DocumentDetailMondrianForm>
+    </Dialog>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from 'vue'
+import { AxiosResponse } from 'axios'
+import { iDocument, iMondrianSchema, iXMLATemplate, iMondrianTemplate } from '../../DocumentDetails'
+import { mapState } from 'vuex'
+import Dialog from 'primevue/dialog'
+import descriptor from './DocumentDetailOlapDesignerDialogDescriptor.json'
+import DocumentDetailXMLAForm from './DocumentDetailXMLAForm.vue'
+import DocumentDetailMondrianForm from './DocumentDetailMondrianForm.vue'
+import Dropdown from 'primevue/dropdown'
+import ProgressSpinner from 'primevue/progressspinner'
+
+const crypto = require('crypto')
+
+export default defineComponent({
+    name: 'document-detail-olap-designer-dialog',
+    components: { Dialog, DocumentDetailXMLAForm, DocumentDetailMondrianForm, Dropdown, ProgressSpinner },
+    props: { visible: { type: Boolean }, selectedDocument: { type: Object as PropType<iDocument> } },
+    emits: ['designerStarted', 'close'],
+    data() {
+        return {
+            descriptor,
+            document: null as iDocument | null,
+            mondrianSchemas: [] as iMondrianSchema[],
+            type: 'mondrian' as string,
+            xmlModel: { address: '', parameters: [] } as iXMLATemplate,
+            mondrianModel: {} as iMondrianTemplate,
+            user: null as any,
+            sbiExecutionId: '',
+            loading: false
+        }
+    },
+    computed: {
+        ...mapState({
+            user: 'user'
+        })
+    },
+    watch: {
+        selectedDocument() {
+            this.loadDocument()
+        }
+    },
+    async created() {
+        this.sbiExecutionId = crypto.randomBytes(16).toString('hex')
+        this.user = (this.$store.state as any).user
```

Comment:
is this variable overriding the one from the mapState computed? Usually it's not a good practice if not needed for timing issues

Reply:
great thx